client/components/Signup.js

```diff
@@ -0,0 +1,99 @@
+import React, {Component} from 'react';
+import { connect } from 'react-redux';
+import { signup } from '../reducers'
+
+
+export class Signup extends Component {
+  constructor() {
+    super()
+    this.state = {
+      firstNameInput: '',
+      lastNameInput: '',
+      emailInput: '',
+      passwordInput: '',
+      dirty: false
+    }
+    this.handleChange = this.handleChange.bind(this)
+  }
+
+  handleChange(evt) {
+    //HANDLE VALIDATIONS?
+    switch (evt.target.name) {
+      case 'firstName':
+        this.setState({firstNameInput: evt.target.value, dirty: true})
+        break;
+      case 'lastName':
+        this.setState({lastNameInput: evt.target.value, dirty: true})
+        break;
+      case 'email':
+        this.setState({emailInput: evt.target.value, dirty: true})
+        break;
+      case 'password':
+        this.setState({passwordInput: evt.target.value, dirty: true})
+        break;
```

Comment:
I THINK I know why you're using `break` instead of `return`, but isn't it fine to use `return` here instead of breaks since we're essentially invoking `handleChange` every time text is entered ?  jw

Reply:
changed